acme_diags/driver/streamflow_driver.py

```diff
@@ -0,0 +1,370 @@
+            var_array = test_data_ts(cdutil.region.domain(latitude=(-90., 90, 'ccb')))
+            if parameter.print_statements:
+                print('test var original dimensions={}'.format(var_array.shape))
+            var_transposed = numpy.transpose(var_array, (2, 1, 0))
+            if parameter.print_statements:
+                print('test var transposed dimensions={}'.format(var_transposed.shape))
+            wrmflow = var_transposed
+            areatotal2 = test_data.get_static_variable('areatotal2', var)
+            area_upstream = numpy.transpose(areatotal2, (1, 0))
+            if parameter.print_statements:
+                print('area_upstream dimensions={}'.format(area_upstream.shape))
+            mask = test_data.get_static_variable('mask', var)
+            mask = numpy.transpose(mask, (1,0))
+            if parameter.print_statements:
+                print('mask dimensions={}'.format(mask.shape))
+        else:
+            data_mat = scipy.io.loadmat(parameter.test_mat_file)
+            if 'E3SMflow' in data_mat.keys():
+                # `edison` file uses this block
+                e3sm_flow = data_mat['E3SMflow']
+                if parameter.print_statements:
+                    print('e3sm_flow = data_mat["E3SMflow"]')
+            else:
+                # `test` file uses this block
+                e3sm_flow = data_mat
+                if parameter.print_statements:
+                    print('e3sm_flow = data_mat')
+            try:
+                if e3sm_flow['uparea'].shape == (1, 1):
+                    # `edison` file uses this block
+                    area_upstream = e3sm_flow['uparea'][0][0]
+                    if parameter.print_statements:
+                        print('e3sm_flow["uparea"] was indexed into for later use')
+                else:
+                    area_upstream = e3sm_flow['uparea']
+                    if parameter.print_statements:
+                        print('e3sm_flow["uparea"] will be used')
+            except KeyError:
+                # `test` file uses this block
+                area_upstream = None
+                if parameter.print_statements:
+                    print('WARNING: uparea not found and will thus not be used')
+            if e3sm_flow['wrmflow'].shape == (1, 1):
+                # `edison` file uses this block
+                wrmflow = e3sm_flow['wrmflow'][0][0]
+                if parameter.print_statements:
+                    print('e3sm_flow["wrmflow"] was indexed into for later use')
+            else:
+                # `test` file uses this block
+                wrmflow = e3sm_flow['wrmflow']
+                if parameter.print_statements:
+                    print('e3sm_flow["wrmflow"] will be used')
+        if parameter.print_statements:
+            print('wrmflow.shape={}'.format(wrmflow.shape))
+
+        # Apply mask
+        # TODO: Appears to do nothing
+        #wrmflow[mask == 2] = numpy.nan
```

Comment:
Applying the `mask` here doesn't seem to change anything. Is there a better way to filter out more gauges?

Reply:
I recall that @hydrotian said we can bump up `max_area_error` to filter out more gauges. But it still is mysterious that with same value of `max_area_error=20`, we get more gauge stations.